Author: Sergey Viuchny <sviuchny@de-bs.ru>

--- nagios-plugins-linux-15.orig/plugins/check_memory.c
+++ nagios-plugins-linux-15/plugins/check_memory.c
@@ -223,7 +223,7 @@ main (int argc, char **argv)
       dpgmajfault = nr_vmem_pgmajfault[1] - nr_vmem_pgmajfault[0];
 
       perfdata_vmem_msg =
-	xasprintf (", vmem_pageins/s=%lu, vmem_pageouts/s=%lu, "
+	xasprintf (" vmem_pageins/s=%lu vmem_pageouts/s=%lu "
 		   "vmem_pgmajfault/s=%lu", dpgpgin, dpgpgout, dpgmajfault);
     }
 
@@ -240,7 +240,7 @@ main (int argc, char **argv)
   status = get_status (mem_percent, my_threshold);
 
   perfdata_memavailable_msg =
-    xasprintf ("mem_available=%Lu%s, ", SU (kb_mem_main_available));
+    xasprintf ("mem_available=%Lu%s ", SU (kb_mem_main_available));
 
   status_msg =
     xasprintf ("%s: %.2f%% (%Lu %s) %s",
@@ -251,10 +251,10 @@ main (int argc, char **argv)
   free (my_threshold);
 
   perfdata_mem_msg =
-    xasprintf ("mem_total=%Lu%s, mem_used=%Lu%s, mem_free=%Lu%s, "
-	       "mem_shared=%Lu%s, mem_buffers=%Lu%s, mem_cached=%Lu%s, %s"
-	       "mem_active=%Lu%s, mem_anonpages=%Lu%s, mem_committed=%Lu%s, "
-	       "mem_dirty=%Lu%s, mem_inactive=%Lu%s"
+    xasprintf ("mem_total=%Lu%s mem_used=%Lu%s mem_free=%Lu%s "
+	       "mem_shared=%Lu%s mem_buffers=%Lu%s mem_cached=%Lu%s %s"
+	       "mem_active=%Lu%s mem_anonpages=%Lu%s mem_committed=%Lu%s "
+	       "mem_dirty=%Lu%s mem_inactive=%Lu%s"
 	       , SU (kb_mem_main_total)
 	       , SU (kb_mem_main_used)
 	       , SU (kb_mem_main_free)
--- nagios-plugins-linux-15.orig/plugins/check_tcpcount.c
+++ nagios-plugins-linux-15/plugins/check_tcpcount.c
@@ -173,10 +173,10 @@ main (int argc, char **argv)
   free (my_threshold);
 
   printf ("%s %s - %lu tcp established | "
-	  "tcp_established=%lu, tcp_syn_sent=%lu, tcp_syn_recv=%lu, "
-	  "tcp_fin_wait1=%lu, tcp_fin_wait2=%lu, tcp_time_wait=%lu, "
-	  "tcp_close=%lu, tcp_close_wait=%lu, tcp_last_ack=%lu, "
-	  "tcp_listen=%lu, tcp_closing=%lu\n",
+	  "tcp_established=%lu tcp_syn_sent=%lu tcp_syn_recv=%lu "
+	  "tcp_fin_wait1=%lu tcp_fin_wait2=%lu tcp_time_wait=%lu "
+	  "tcp_close=%lu tcp_close_wait=%lu tcp_last_ack=%lu "
+	  "tcp_listen=%lu tcp_closing=%lu\n",
 	  program_name_short, state_text (status), tcp_established,
 	  tcp_established, tcp_syn_sent, tcp_syn_recv,
 	  tcp_fin_wait1, tcp_fin_wait2, tcp_time_wait,
--- nagios-plugins-linux-15.orig/plugins/check_swap.c
+++ nagios-plugins-linux-15/plugins/check_swap.c
@@ -190,7 +190,7 @@ main (int argc, char **argv)
 			  percent_used, SU (kb_swap_used));
 
   perfdata_swap_msg =
-    xasprintf ("swap_total=%Lu%s, swap_used=%Lu%s, swap_free=%Lu%s, "
+    xasprintf ("swap_total=%Lu%s swap_used=%Lu%s swap_free=%Lu%s "
 	       /* The amount of swap, in kB, used as cache memory */
 	       "swap_cached=%Lu%s", SU (kb_swap_total), SU (kb_swap_used),
 	       SU (kb_swap_free), SU (kb_swap_cached));
